File: agent/commands-generator/src/main/java/org/matilda/commands/CommandsGenerator.java
package org.matilda.commands;

import org.matilda.commands.collectors.ServicesCollector;
import org.matilda.commands.info.ServiceInfo;
import org.matilda.commands.processors.Processor;

import javax.inject.Inject;

public class CommandsGenerator {
    @Inject
    ServicesCollector mServicesCollector;

    @Inject
    Processor<ServiceInfo> mServiceProcessor;

    @Inject
    CommandsGenerator() {}

    public void generate() {
        mServicesCollector.collect().processEachService(mServiceProcessor);
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/CommandsGenerator.kt
package org.matilda.commands

import org.matilda.commands.collectors.ServicesCollector
import org.matilda.commands.info.ProjectServices
import org.matilda.commands.processors.Processor
import javax.inject.Inject

class CommandsGenerator @Inject constructor() {
    @Inject
    lateinit var mServicesCollector: ServicesCollector

    @Inject
    lateinit var mProjectServicesProcessor: Processor<ProjectServices>

    fun generate() {
        val services = mServicesCollector.collect()
        mProjectServicesProcessor.process(services)
    }
}
